pkg/controller/vsphere/actuator_test.go

```diff
@@ -0,0 +1,284 @@
+package vsphere
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"path/filepath"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	machinev1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	vsphereapi "github.com/openshift/machine-api-operator/pkg/apis/vsphereprovider/v1beta1"
+	"github.com/vmware/govmomi/simulator"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+func init() {
+	// Add types to scheme
+	machinev1.AddToScheme(scheme.Scheme)
+	configv1.AddToScheme(scheme.Scheme)
+}
+
+func TestMachineEvents(t *testing.T) {
+	g := NewWithT(t)
+
+	// Setup vsphere test environment
+	model, session, server := initSimulator(t)
+	defer model.Remove()
+	defer server.Close()
+
+	host, port, err := net.SplitHostPort(server.URL.Host)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	credentialsSecretUsername := fmt.Sprintf("%s.username", host)
+	credentialsSecretPassword := fmt.Sprintf("%s.password", host)
+
+	password, _ := server.URL.User.Password()
+	namespace := "test"
+	vm := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
+
+	testEnv := &envtest.Environment{
+		CRDDirectoryPaths: []string{
+			filepath.Join("..", "..", "..", "install"),
+			filepath.Join("..", "..", "..", "vendor", "github.com", "openshift", "api", "config", "v1")},
+	}
+
+	// Setup k8s test environment
+	cfg, err := testEnv.Start()
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(cfg).ToNot(BeNil())
+	defer func() {
+		g.Expect(testEnv.Stop()).To(Succeed())
+	}()
+
+	mgr, err := manager.New(cfg, manager.Options{
+		Scheme:             scheme.Scheme,
+		MetricsBindAddress: "0",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	doneMgr := make(chan struct{})
+
+	go func() {
+		g.Expect(mgr.Start(doneMgr)).To(Succeed())
```

Comment:
Need a GinkgoRecover for go routines
```suggestion
		defer GinkgoRecover()
		g.Expect(mgr.Start(doneMgr)).To(Succeed())
```

Reply:
Ignore me, not using ginkgo here